Index: simplepie.inc
===================================================================
--- simplepie.inc	(revision 754)
+++ simplepie.inc	(working copy)
@@ -6585,6 +6585,11 @@
 				curl_setopt($fp, CURLOPT_REFERER, $url);
 				curl_setopt($fp, CURLOPT_USERAGENT, $useragent);
 				curl_setopt($fp, CURLOPT_HTTPHEADER, $headers2);
+
+				// added by FoF to enable https and digest authentication
+				curl_setopt($fp, CURLOPT_SSL_VERIFYPEER, false);
+				curl_setopt($fp, CURLOPT_HTTPAUTH, CURLAUTH_ANY);
+
 				if (!ini_get('open_basedir') && !ini_get('safe_mode'))
 				{
 					curl_setopt($fp, CURLOPT_FOLLOWLOCATION, 1);
