Index: simplepie.inc
===================================================================
--- simplepie.inc	(revision 802)
+++ simplepie.inc	(working copy)
@@ -6497,6 +6497,10 @@
 					curl_setopt($fp, CURLOPT_FOLLOWLOCATION, 1);
 					curl_setopt($fp, CURLOPT_MAXREDIRS, $redirects);
 				}
+                
+				// added by FoF to enable https and digest authentication
+				curl_setopt($fp, CURLOPT_SSL_VERIFYPEER, false);
+				curl_setopt($fp, CURLOPT_HTTPAUTH, CURLAUTH_ANY);
 
 				$this->headers = curl_exec($fp);
 				if (curl_errno($fp) == 23 || curl_errno($fp) == 61)
